src/narm_hypervisor.rs

```diff
@@ -298,6 +298,79 @@ impl NarmHypervisor {
                     }
                 }
 
+                //********************************//
+                //**   Context info operators   **//
+                //********************************//
+
+                //SVC 0x90: gas_remaining() -> limit: u64
+                //Get the amount of gas left and return
+                0x90 => {
+                    let limit = codata.gas_remaining;
+
+                    let limit_least_significant = (limit & 0x0000_0000_FFFF_FFFF) as u32;
+                    let limit_most_significant = ((limit & 0xFFFF_FFFF_0000_0000) >> 32) as u32;
+
+                    // VM is little endian, so we put them in that order
+                    self.vm.external_set_reg(0, limit_least_significant);
+                    self.vm.external_set_reg(1, limit_most_significant);
+                }
+
+                //SVC 0x91: self_address() -> address: stack NeutronAddress
+                //Get the address of the current execution and push to stack
+                0x91 => {
+                    // TODO: replace with something more efficient
+                    let address = &codata.current_context().self_address;
+                    let mut bytes = address.version.to_le_bytes().to_vec();
+                    bytes.append(&mut address.data.to_vec());
+
+                    match codata.push_output_stack(&bytes) {
+                        Ok(_) => {}
+                        Err(e) => {
+                            return Ok(HypervisorState::Error(e));
+                        }
+                    }
+                }
+
+                //SVC 0x92: origin() -> address: stack NeutronAddress
+                //Get the address that caused the current chain of executions and push to stack
+                0x92 => {
+                    // TODO: replace with something more efficient
+                    let address = &codata.current_context().origin;
+                    let mut bytes = address.version.to_le_bytes().to_vec();
+                    bytes.append(&mut address.data.to_vec());
+
+                    match codata.push_output_stack(&bytes) {
```

Comment:
Would it make sense to add a helper function for this whole operation? ie, having an extension function so that its possible to do `codata.push_output_stack_address` ?

Reply:
I'd say it should be a function of the NeutronAddress struct. It already has one to construct an instance from raw bytes, it would only make sense to have a function that goes the other way too.

Reply:
Fair enough, as long as neutron-common doesn't end up depending on neutron-host/neutron-star

Reply:
That also keeps the data processing code as close as possible to the actual data structure, which is good maintenance-wise. 

In the code that runs in the VM I just coerce the entire struct to bytes, but that only works when you can assume LE byte order.